src/features/groups/list/index.tsx

```diff
@@ -22,35 +28,127 @@ const filterTabs: IFilterTab[] = [
 
 const GroupsList = () => {
   const { t } = useTranslation('groups');
+  const groups = useSelector(groupsListSelector);
+  const dispatch = useDispatch();
 
-  const { data: groups } = useGetTransactionGroupsQuery();
+  const [loadGroups] = useLazyGetTransactionGroupsQuery();
 
   const [activeTab, setActiveTab] = useState<IFilterTab>(filterTabs[0]);
 
+  const [isGroupDetailsMode, setIsGroupDetailsMode] = useState<boolean>(false);
+  const [isAddGroupMembersMode, setIsAddGroupMembersMode] = useState<boolean>(false);
+  const [editedGroupId, setEditedGroupId] = useState<string | null>(null);
+
   const filteredGroups = useMemo(() => activeTab.filter(groups || []), [
     groups, activeTab
   ]);
 
   const onTabClick = useCallback((key: string) => {
     setActiveTab(filterTabs.find(c => c.key === key)!);
+  }, [setActiveTab]);
+
+  const onAddIconClick = useCallback(() => {
+    setIsGroupDetailsMode(true);
+  }, [setIsGroupDetailsMode]);
+
+  const onBackIconClick = useCallback(() => {
+    if (isAddGroupMembersMode) {
+      setIsAddGroupMembersMode(false);
+    } else if (isGroupDetailsMode) {
+      setIsGroupDetailsMode(false);
+      setEditedGroupId(null);
+    }
+  }, [isAddGroupMembersMode, isGroupDetailsMode]);
+
+  const onGroupSaved = useCallback((group: ITransactionGroup) => {
+    setIsGroupDetailsMode(false);
+    setEditedGroupId(null);
+
+    dispatch(saveGroup(group));
+  }, [dispatch]);
+
+  const onAddMembersModeChange = useCallback((value: boolean) => {
+    setIsAddGroupMembersMode(value);
+  }, []);
+
+  const onGroupClick = useCallback((group: ITransactionGroupListItem) => {
+    setEditedGroupId(group.id);
+    setIsGroupDetailsMode(true);
   }, []);
 
+  const headerText = () => {
+    if (isAddGroupMembersMode) {
+      return t('groupMembers');
+    }
+
+    if (isGroupDetailsMode) {
+      return editedGroupId ? t('updateGroup') : t('createGroup');
+    }
+
+    return t('groups');
```

Comment:
You can use `useMemo` and make `headerText` not function but already selected text